src/query.rs

```diff
@@ -290,16 +287,30 @@ pub struct Witness {
 }
 
 /// A [`trustfall`] query, containing the query string itself
-/// and a mapping of argument names to [`TransparentValue`]s provided
-/// to the query.
+/// and a mapping of argument names to value types which are
+/// provided to the query.
 #[derive(Debug, Clone, Serialize, Deserialize)]
 pub struct Query {
     /// The string containing the Trustfall query.
     pub query: String,
 
     /// The mapping of argument names to values provided to the query.
+    ///
+    /// These can be inherited from a previous query ([`InheritedValue::Inherited`]) or
+    /// specified as [`InheritedValue::Constant`]s.
     #[serde(default)]
-    pub arguments: BTreeMap<String, TransparentValue>,
+    pub arguments: BTreeMap<String, InheritedValue>,
+}
+
+/// Represents either a value inherited from a previous query, or a
+/// provided constant value.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub enum InheritedValue {
+    /// Inherit the value from the previous output whose name is the given `String`.
+    Inherited(String),
+    /// Provide the constant value specified here.
+    #[serde(untagged)]
+    Constant(TransparentValue),
 }
```

Comment:
Excellent, this is exactly what I was hoping for.

Reply:
never mind :sweat_smile: it seems like `ron` has known limitations with `untagged`/`deserialize_any` and having nested `untagged` makes it so the deserialization doesn't work :pensive:.  Maybe we have a list (or map of new_name -> old_name) of `inherit_arguments` along with a map of `arguments`?  Then it's an error if the maps share keys.  Alternately, we could remove the `untagged` so users would have to express arguments as `Inherit("...")` or `Constant(...)`, which prevents duplicate keys from being expressed but is more verbose and less consistent with the `arguments` top-level field.

Reply:
Oof, that's unfortunate 😞 What do you think is the better choice between the list/renaming map and the explicit `Inherit(..)` vs `Constant(..)`?

If you feel it's worth writing a few more witness sections before deciding, that's fine by me. We can go the easier route now and revisit based on your experience of using it in the real world.

Reply:
I think I would prefer the explicit one.  I also thought of another idea that might work.  It's more jsony than rusty but it does remove the requirement to specify `Constant(...)`:

```ron
arguments: {
  "abc": (inherit: "abc"),
  "string": "literal_string",
  "int": -30,
}
```

where we specify inheriting as a one-field struct `(inherit: "name")` or verbosely, `Inherited(inherit: "name")`.  How do you feel about this.  It won't have any conflicts (I think) with `TransparentValue` because maps and structs in `ron` have different syntax, so `Inherit` and `Constant` variants of the enum should be disjoint.

Reply:
I pushed a test that shows that the deserialization is what we expect.